### ISSUE TITLE
Feature request: Keycloak integration

### ISSUE THREAD

**Issue by NGroens** [Intermesh/groupoffice#890]
Hello,

are there also plans for other OAuth providers like a self-hosted Keycloak server?
For this, it would be necessary to be able to define the entry points yourself.

Thanks.

**Comment by michalcharvat** [comment]
Well there is https://github.com/stevenmaguire/oauth2-keycloak package so it could be possible I guess.

**Comment by michalcharvat** [comment]
I guess in this case would be the best extend oauth2client module by another provider (for sure the settings table has to be extended by additional details - perhaps json column where you could copy the whole configuration - keycloak offers json format similar to google).

<img width="620" alt="image" src="https://user-images.githubusercontent.com/8058273/195211031-d705af7a-f4dc-4c56-ac6c-80e84fa3884d.png">

After that the major step should come - create oauth authenticator module @mschering 

---
https://docs.axway.com/bundle/EBICSClient_11_allOS_en_HTML5/page/install_and_configure_keycloak_server.html
https://www.keycloak.org/getting-started/getting-started-docker

**Comment by derjoachim** [comment]
@NGroens, what is your use case? For now, I am assuming Single SignOn, but please feel free to specify. We currently only use the OAuth2 client to connect to certain IMAP providers, so in its current form, just having the OAuth2 connection would be of limited use. A way to use said connection would have to be implemented as well. SSO would be awesome, but currently our focus is on several other projects.

**Comment by NGroens** [comment]
@derjoachim Yes exactly. I assumed that the client can be used for single sign on. If this is not the case, then Keycloak is indeed not necessary, and I misunderstood.
I would be very happy to have single sign on. But I can also understand if there are other, more important projects.

**Comment by mschering** [comment]
We can make an offer to implement this if you like.

**Comment by NGroens** [comment]
Thank you for the offer. However, this feature is not so important for me. I was just interested if this function is planned.

**Comment by pschiffe** [comment]
This seems to be possible now with OIDC module.
